engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetInputTextureFromFbo.java

```diff
@@ -103,16 +103,7 @@ public SetInputTextureFromFbo(int textureSlot, FBO fbo, FboTexturesTypes texture
      */
     public SetInputTextureFromFbo(int textureSlot, SimpleUri fboUri, FboTexturesTypes textureType, BaseFBOsManager fboManager,
                                   ResourceUrn materialUrn, String shaderParameterName) {
-        this.textureSlot = textureSlot;
-        this.textureType = textureType;
-        this.fbo = fboManager.get(fboUri);
-        this.materialUrn = materialUrn;
-        this.shaderParameterName = shaderParameterName;
-
-        this.material = getMaterial(materialUrn);
-
-        propertyChange(null); // Cheeky way to initialise textureId
-        fboManager.subscribe(this);
+        this(textureSlot, fboManager.get(fboUri), textureType, fboManager, materialUrn, shaderParameterName);
```

Comment:
As I look at this, and the lack of checks on the arguments, I wonder how defensive would be good to be when we open the API to modules. Let's cross that bridge when we get there, but my point here is that a custom node might use this and other state change objects, and if they are used incorrectly, i.e. passing wrong or null arguments, we might want to make sure the developer gets some good error logging about it.